src/tikv.md

```diff
@@ -0,0 +1,212 @@
+- The cop task has to wait for worker resources and snapshot fetch:
+  - The cop task wait schedule duration is observed as `tikv_coprocessor_request_wait_seconds{type="schedule"}`. Also this duration is
+    recorded in the request tracker as `schedule_wait_time`.
+  - The The cop handler build time is observed as `tikv_coprocessor_request_handler_build_seconds{type="req"}`. Also it's recorded in the request 
+    tracker as `handler_build_time`.
+  - The total wait time is observed as `tikv_coprocessor_request_wait_seconds{type="all"}`
+- The cop task execution time is observed as `tikv_coprocessor_request_handle_seconds{type="req"}`. Also is's recorded in the request tracker as 
+  `req_lifetime`
+  
+TODO: Build a detailed model describing the relationship bettween the user perceivable durations and the information recorded in the tracker. Then
+it would be straightforward to tell the duration combinations of `cop scheduler`, `task wait`, `io operations like scan`.
+  
+
+### Snapshot Fetch
+
+```railroad
+Diagram(
+  Choice(
+    0,
+    Sequence(
+      NonTerminal("Local Read"),
+    ),
+    Sequence(
+      NonTerminal("Propose Wait"),
+       Sequence(
+         NonTerminal("Read index Read Wait"),
+      ),
+    ),
+  ),
+  Sequence(
+      NonTerminal("Fetch A Snapshot From KV Engine"),
+  ),
+)
+```
+
+- The number of local read requests rejected is observed as `tikv_raftstore_local_read_reject_total`. If most of the requests are rejected by local reader the
+  performance would be bad.
+- The duration of the read index wait could be regarded as `tikv_raftstore_commit_log_duration_seconds_bucket`. 
+
+TODO: Maybe it's useful to record the read index wait duration in the tracker for slow query diagnose.
+
+
+## TxnKV Write
+
+```railroad
+Diagram(
+  Sequence(
+      NonTerminal("Grpc Receive"),
+  ),
+  Sequence(
+    NonTerminal("Acquire Latch For Keys"),
+  ),
+  Choice(
+    0,
+    Sequence(
+      NonTerminal("Get Snapshot From KV Engine Local Read"),
+    ),
+    Sequence(
+      NonTerminal("Get Snapshot From KV Engine Leader Confirm"),
+    ),
+  ),
+  
+  Choice(
+    0,
+    NonTerminal("Process Write Requests"),
+  ),
+  Choice(
+    0,
+    NonTerminal("Async Write"),
+  ),
+  Sequence(
+      NonTerminal("Grpc Response"),
+  ),
+)
+```
+
+- The latch acquisition is observed as `TiKV_scheduler_latch_wait_duration_seconds{type=xxx}`.
+- The write command processing may needs to read related from engine first.
+- The async write stage includes both transaction log consistency and state machine apply, it's observed as `tikv_storage_engine_async_request_duration_seconds{type=write}`.
+
+
+
+### TxnKV Async Write
+
+``` railroad
+Diagram(
+  Sequence(
+      NonTerminal("Propose Wait"),
+  ),
+  Choice(
+    0,
+    Sequence(
+      NonTerminal("Append Log"),
+    ),
+    Sequence(
+      NonTerminal("Commit Log Wait"),
+    ),
```

Comment:
In the snapshot-fetch section, the branches stand for conditions, here we use branches for parallelism, maybe there is a better way to denote parallel tasks. @zyguan what's your idea?